### ISSUE TITLE
How about CI?

### ISSUE THREAD

**Issue by cheako** [TheCherno/Hazel#74]
I've used travis-ci.org with success, though I think Windows support is a no-go.  I would be surprised, given the massive market share, that there isn't a solution...  Google says AppVeyor, so there may yet be ppl who use continue to use Windows.

**Comment by Gaztin** [comment]
I've used AppVeyor in other projects and it is very fast and intuitive. And most importantly it won't take long to set up. Thank you for bringing it up! I'll try to get it up and running soon.

**Comment by Gaztin** [comment]
I've added AppVeyor, Future pull requests should have to pass the compilation from now on!

**Comment by Gaztin** [comment]
Looks like it's not adding checks for PRs. I think @TheCherno has to make some changes in the settings, but not sure what's blocking it. Everything looks good in the AppVeyor project.

**Comment by LovelySanta** [comment]
I just made PR #79 and it did not add any checks to it.

However shields.io is saying: ![shield](https://img.shields.io/appveyor/ci/Gaztin/Hazel.svg), but that's on Gaztin's appveyor account.

Cherno might have to add appveyor for it to show up on the actual PR's itself.
As of now, github still does not show any checks.

**Comment by LovelySanta** [comment]
If you look in the commits; it did check cherno adding new content (not through PR), but only one of the two...
![image](https://user-images.githubusercontent.com/26593477/60982089-c4b16680-a337-11e9-81ab-8cd416097199.png)

**Comment by TheCherno** [comment]
My ultimate plan was to use GitLab for all of our CI needs (@Gaztin you should know about this). When I get back to Melbourne I'll take a proper look at all this and make sure we have what we need setup, including the appropriate branches.

**Comment by Gaztin** [comment]
You're right. I did know. I was simply living up to my reputation of having the *worst* memory. Thanks for clearing it up. I'll remove the AppVeyor as soon as the GitLab one is in place. At least we now have something that shows us if a merged PR is invalid!

**Comment by L4ZZA** [comment]
> My ultimate plan was to use GitLab for all of our CI needs (@Gaztin you should know about this). When I get back to Melbourne I'll take a proper look at all this and make sure we have what we need setup, including the appropriate branches.

Does that mean that we will all have to port the repo from github to gitlab? How does GitLab CI handle Windows build I can't find any straight forward way to run a pipeline on GitLab CI like we do on AppVeyor were windows and VS support come out of the box.

**Comment by LovelySanta** [comment]
Since then github actions are introduced, and they do also support CI. I'm not sure how viable this is, but just throwing this on the table, since it wasn't an option before.

**Comment by Gaztin** [comment]
Best case, you can self-host the GitLab runner on a dedicated server computer. That way you have total control over the environment and build tools. But that said, I don't see why you wouldn't be able to run MSBuild through the public GitLab CI?

**Comment by L4ZZA** [comment]
All I'm saying is that in order to run on GitLab CI/CD a copy of the repository is required, it will be mirrored so technically there's no need to maintain it, but I believe there are better options. 

I do quite like the way GitLab CI is set up, but is not straight forward to create a windows/visual-studio build like it is on AppVeyor. 
What I don't like about AppVeyor tho, is the documentation and the small, not very active community behind it.

**Comment by caelwithcats** [comment]
GitHub actions provides good CI tools. It can compile C++ for windows, Linux and mac. Why not use that?
